app/controllers/application_controller.rb

```diff
@@ -6,10 +6,17 @@ class ApplicationController < ActionController::Base
   before_action :authenticate_user!
   before_action :configure_permitted_parameters, if: :devise_controller?
 
-  protected
+  rescue_from ActiveRecord::RecordInvalid, with: :render_error
+
+  private
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
     devise_parameter_sanitizer.permit(:account_update, keys: [:name])
   end
+
+  def render_error(exception)
+    flash[:alert] = exception
+    redirect_to new_category_path
+  end
 end
```

Comment:
Just as it, but in future  how render new category template without using flash, beacuse I've already have error partial.
`render :new
or
renedr '/new'`
fails (
Also if I add transaction for edit action, how to define what to render new or edit template

Reply:
if you want redirect user to new path use 
redirect_to 
but if you want set flash message - it needed be changed